### PR TITLE
fix: improve typing of `Messageable.send` and http methods

### PR DESCRIPTION
**Pull request by shiftinv**
## Summary

This PR adds annotations to the implementation of `Messageable.send` (previously, only the overloads had annotations), makes the `tts` variable non-optional, and fixes the annotations of the `nonce`, `embeds` and `stickers` parameters.

## Checklist

- [x] If code changes were made, then they have been tested
    - [ ] I have updated the documentation to reflect the changes
    - [x] I have formatted the code properly by running `task lint` or `pre-commit run --all-files`
- [x] This PR fixes an issue
- [ ] This PR adds something new (e.g. new method or parameters)
- [ ] This PR is a breaking change (e.g. methods or parameters removed/renamed)
- [ ] This PR is **not** a code change (e.g. documentation, README, ...)

### CODE REVIEW

File: disnake/abc.py [hunk 1]
```diff
@@ -1296,21 +1296,21 @@ async def send(
 
     async def send(
         self,
-        content=None,
+        content: Optional[Any] = None,
```
Comment (EQUENOS) [inline disnake/abc.py:1299]: `Optional[Any]` is `Any`

**Review by EQUENOS — APPROVED** [review]
lgtm, small nitpick:

File: disnake/abc.py (hunk 1 above)
Comment (shiftinv) [reply disnake/abc.py:1299]: Not necessarily:
```py
from typing import Any, Optional, cast

x = cast(Optional[Any], 42)
y: int = x    # error: Incompatible types in assignment (expression has type "Optional[Any]", variable has type "int")
```

The overloads also use `Optional[Any]`, which is why I used it there as well

Comment (EQUENOS) [reply disnake/abc.py:1299]:
```python
from typing import Any, Optional

def foo(arg: Any): ...

def bar(arg: Optional[Any]): ...

foo(42)
foo(None)

bar(42)
bar(None)

# both ways work
```

Comment (shiftinv) [reply disnake/abc.py]: 7e002c99d21d24681aaf20a341395cbd93a150b5

**Pull request merged**